scene/main/scene_tree_fti.cpp

```diff
@@ -318,12 +318,16 @@ void SceneTreeFTI::_update_dirty_nodes(Node *p_node, uint32_t p_current_frame, f
 		return;
 	}
 
+	p_node->update_cached_children();
+	Span<Node *> children = p_node->get_cached_children();
+	uint32_t num_children = children.size();
+
 	// Not a Node3D.
 	// Could be e.g. a viewport or something
 	// so we should still recurse to children.
 	if (!s) {
-		for (int n = 0; n < p_node->get_child_count(); n++) {
-			_update_dirty_nodes(p_node->get_child(n), p_current_frame, p_interpolation_fraction, p_active, nullptr, p_depth + 1);
+		for (uint32_t n = 0; n < num_children; n++) {
```

Comment:
It may be easier to just iterate the `Span`

Reply:
Pretty much the same and I hate iterator syntax, find traditional is easier to read and follow what the compiler is doing. :grin: